### PR TITLE
Add error page for exception handling.

### PR DESCRIPTION
**Pull request by ChamathNS**
### Proposed changes in this pull request

$subject.

### When should this PR be merged

Soon

### CODE REVIEW

File: io.asgardio.tomcat.saml.agent.sample/src/main/webapp/error.jsp [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+<body>
+<main class="center-segment">
+    <div class="element-padding">
+        <div class="element-padding">
+            <%=exception.getMessage()%>
+        </div>
+    </div>
+</main>
+</body>
+</html>
```
Comment (gayashanbc) [inline io.asgardio.tomcat.saml.agent.sample/src/main/webapp/error.jsp:60]: Add a newline at the EOF

Comment (ChamathNS) [reply io.asgardio.tomcat.saml.agent.sample/src/main/webapp/error.jsp]: Resolved with https://github.com/asgardio/asgardio-tomcat-saml-agent/pull/19/commits/0d0400aff77df1a82b658a3d459a6bbc83469897.

**Review by gayashanbc — APPROVED** [review]

**Comment by darshanasbg** [comment]
This way, we are not handling this agent exception, rather let the web server to handle it.. We should handle the exceptions from related code blocks (may be from the valve) without letting this agent exceptions to get passed to web server level.

**Comment by ChamathNS** [comment]
> This way, we are not handling this agent exception, rather let the web server to handle it.. We should handle the exceptions from related code blocks (may be from the valve) without letting this agent exceptions to get passed to web server level.

If we are handling it in the agent level then, we may need to add a new property to identify the error page, wouldn't we?

**Comment by ChamathNS** [comment]
> This way, we are not handling this agent exception, rather let the web server to handle it.. We should handle the exceptions from related code blocks (may be from the valve) without letting this agent exceptions to get passed to web server level.

Resolved with https://github.com/asgardio/asgardio-tomcat-saml-agent/pull/19/commits/eca3fd8fbdb95dd582e8141ff2a9b33011f429f5/.

File: io.asgardio.tomcat.saml.agent.sample/src/main/resources/sample-app.properties [hunk 2]
```diff
@@ -28,6 +28,8 @@ SAML2.SPEntityId=sample-app
 
 IndexPage=/sample-app/index.html
 
+ErrorPage=/error.jsp
```
Comment (darshanasbg) [inline io.asgardio.tomcat.saml.agent.sample/src/main/resources/sample-app.properties:31]: do we need '/'

**Review by darshanasbg — APPROVED** [review]

File: io.asgardio.tomcat.saml.agent.sample/src/main/resources/sample-app.properties (hunk 2 above)
Comment (ChamathNS) [reply io.asgardio.tomcat.saml.agent.sample/src/main/resources/sample-app.properties:31]: Yes. If we are not adding it here, we'd have to do it in code then.

**Review by darshanasbg — APPROVED** [review]

**Pull request merged**